src/basic/terminal-util.c

```diff
@@ -1267,11 +1267,17 @@ int vt_verify_kbmode(int fd) {
 }
 
 int vt_reset_keyboard(int fd) {
-        int kb;
+        int kb, r;
 
         /* If we can't read the default, then default to unicode. It's 2017 after all. */
         kb = vt_default_utf8() != 0 ? K_UNICODE : K_XLATE;
 
+        r = vt_verify_kbmode(fd);
+        if (r < 0) {
+                log_debug_errno(r, "Keyboard is not in XLATE or UNICODE mode, not resetting: %m");
+                return r;
```

Comment:
log_debug_errno() returns the error code you passed in, so that you can shorten this to:

```c
if (r < 0)
        return log_debug_errno(r, …
```

Reply:
@poettering should not we eat up the error here, too, instead?
Having VT in raw mode is pretty normal and users may see warnings like this:
systemd-logind[1090]: Failed to restore VT, ignoring: Device or resource busy

Reply:
@poettering I added eating up the error about the keyboard mode in the last update.